packages/analytics-js/src/components/userSessionManager/UserSessionManager.ts

```diff
@@ -51,10 +62,47 @@ class UserSessionManager implements IUserSessionManager {
       this.setInitialReferrer(referrer);
       this.setInitialReferringDomain(getReferringDomain(referrer));
     }
+    // Initialize session tracking
+    this.initializeSessionTracking();
     // Register the effect to sync with storage
     this.syncSessionWithStorage();
   }
 
+  /**
+   * A function to initialize sessionTracking
+   */
+  initializeSessionTracking() {
+    const sessionInfo: SessionInfo = this.getSession() || defaultSessionInfo;
+
+    let finalAutoTrackingStatus = !(
+      state.loadOptions.value.sessions.autoTrack === false || sessionInfo.manualTrack === true
+    );
+
+    const sessionTimeout: number = state.loadOptions.value.sessions.timeout as number;
+
+    if (isNumber(sessionTimeout)) {
+      if (sessionTimeout === 0) {
+        this.logger?.warn(
+          '[SessionTracking]:: Provided timeout value 0 will disable the auto session tracking feature.',
+        );
+        finalAutoTrackingStatus = false;
+      }
+      // In case user provides a setTimeout value greater than 0 but less than 10 seconds SDK will show a warning
+      // and will proceed with it
+      if (sessionTimeout > 0 && sessionTimeout < MIN_SESSION_TIMEOUT) {
+        this.logger?.warn(
+          '[SessionTracking]:: It is not advised to set "timeout" less than 10 seconds',
+        );
+      }
+      sessionInfo.timeout = sessionTimeout;
+    }
+    state.session.sessionInfo.value = { ...sessionInfo, autoTrack: finalAutoTrackingStatus };
+    // If auto session tracking is enabled start the session tracking
+    if (state.session.sessionInfo.value.autoTrack) {
```

Comment:
should this be an effect to invoke startAutoTracking whenever the state.session.sessionInfo.value.autoTrack value changes to true?

Reply:
The function has been modified. Now it is used to start or renew auto-tracking and does not depend on `state.session.sessionInfo.value.autoTrack=false`
Not ideal for using effect.